stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/viewmodel/messages/MessageListViewModel.kt

```diff
@@ -508,13 +519,30 @@ public class MessageListViewModel(
                 previousUser == user && nextUser != user -> Bottom
                 else -> None
             }
-            MessageItem(
-                message = message,
-                groupPosition = position,
-                parentMessageId = parentMessageId,
-                isMine = user.id == currentUser?.id
+
+            if (shouldAddDateSeparator(previousMessage, message)) {
+                groupedMessages.add(DateSeparator(message.getCreatedAtOrThrow()))
+            }
+
+            groupedMessages.add(
+                MessageItem(
+                    message = message,
+                    groupPosition = position,
+                    parentMessageId = parentMessageId,
+                    isMine = user.id == currentUser?.id
+                )
             )
-        }.reversed()
+        }
+
+        return groupedMessages.reversed()
+    }
+
+    private fun shouldAddDateSeparator(previousMessage: Message?, message: Message): Boolean {
+        return if (previousMessage == null) {
+            true
+        } else {
+            (message.getCreatedAtOrThrow().time - previousMessage.getCreatedAtOrThrow().time) > (1000 * 60 * 60 * 4)
```

Comment:
```suggestion
            (message.getCreatedAtOrThrow().time - previousMessage.getCreatedAtOrThrow().time) > TimeUnit.HOURS.toMillis(4)
```
Maybe it'd be more readable like this?

Reply:
Yeah, great suggestion - I just took the implementation from the UI Comp :D I'll test this and see that it works and then update!